### PR TITLE
 Add function withoutDuplicates at extension Sequence

### PR DESCRIPTION
**Pull request by saucym**
## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 4.
- [x] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

File: Sources/SwifterSwift/SwiftStdlib/SequenceExtensions.swift [hunk 1]
```diff
@@ -154,6 +154,19 @@ public extension Sequence {
...
 
+    /// SwifterSwift: Remove duplicate elements based on condition.
+    ///
+    ///        [1, 2, 1, 3, 2].withoutDuplicates { $0 } -> [1, 2, 3]
+    ///        [(1, 4), (2, 2), (1, 3), (3, 2), (2, 1)].withoutDuplicates { $0.0 } -> [(1, 4), (2, 2), (3, 2)]
+    ///
+    /// - Parameter condition: conditions to evaluate repeating elements.
+    /// - Returns: Sequence without repeating elements
+    /// - Complexity: O(*n*), where *n* is the length of the sequence.
+    func withoutDuplicates<E: Hashable>(by condition: (Element) -> E ) -> [Element] {
```
Comment (guykogus) [inline Sources/SwifterSwift/SwiftStdlib/SequenceExtensions.swift]: 1. In order to be in line with the other functions, let's make the closure throwable.
2. I don't think that `condition` is the right name for the parameter. You're transforming the element into some hashable value, so `transform` makes more sense to me.
3. `T` is the default generic parameter, unless you want to indicate something more specific. Using `E` confuses it a bit with `Element`, but that's not what `E` represents.
```suggestion
    func withoutDuplicates<T: Hashable>(transform: (Element) throws -> T) rethrows -> [Element] {
```
You'll also need to change the call below to
```Swift
return try filter { !set.insert(try transform($0)).inserted }
```

**Comment by guykogus** [comment]
Please note that your tests are failing on Linux. @LucianoPAlmeida can you please help with that? Any idea why it's failing?

**Comment by guykogus** [comment]
Please note that your tests are failing on Linux because you've used `IndexPath` in your tests, which are part of `Foundation`, not the standard library. Please use a different `struct`/`class`. Don't import `Foundation`.

File: Tests/SwiftStdlibTests/SequenceExtensionsTests.swift [hunk 2]
```diff
@@ -85,6 +85,11 @@ final class SequenceExtensionsTests: XCTestCase {
         XCTAssertThrowsError(try [2].single(where: { _ in throw SequenceTestError.closureThrows }))
     }
 
+    func testWithoutDuplicates() {
+        XCTAssertEqual([1, 2, 1, 3, 2].withoutDuplicates { $0 }, [1, 2, 3])
+        XCTAssertTrue([[1, 4], [2, 2], [1, 3], [3, 2], [2, 1]].withoutDuplicates { $0[0] } == [[1, 4], [2, 2], [3, 2]])
```
Comment (guykogus) [inline Tests/SwiftStdlibTests/SequenceExtensionsTests.swift]:
```suggestion
        XCTAssertEqual([[1, 4], [2, 2], [1, 3], [3, 2], [2, 1]].withoutDuplicates { $0[0] }, [[1, 4], [2, 2], [3, 2]])
```

File: Sources/SwifterSwift/SwiftStdlib/SequenceExtensions.swift [hunk 3]
```diff
@@ -154,6 +154,19 @@ public extension Sequence {
         return singleElement
     }
 
+    /// SwifterSwift: Remove duplicate elements based on condition.
+    ///
+    ///        [1, 2, 1, 3, 2].withoutDuplicates { $0 } -> [1, 2, 3]
+    ///        [(1, 4), (2, 2), (1, 3), (3, 2), (2, 1)].withoutDuplicates { $0.0 } -> [(1, 4), (2, 2), (3, 2)]
+    ///
+    /// - Parameter transform: transform to evaluate repeating elements.
```
Comment (LucianoPAlmeida) [inline Sources/SwifterSwift/SwiftStdlib/SequenceExtensions.swift]:
```suggestion
    /// - Parameter transform: A closure that should return the value to be evaluated on repeating elements.
```
What you think?

**Comment by SD10** [comment]
Thank you for contributing to SwifterSwift! I've invited you to join the SwifterSwift GitHub organization - no pressure to accept! If you'd like more information on what that means, check out our [contributing guidelines](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md). Feel free to reach out if you have any questions! 😃

**Pull request merged**